src/action/product.js

```diff
@@ -0,0 +1,27 @@
+import {
+  END_DATA_PRODUCT_DETAILS,
+  CHOOSE_IMG,
+  START_DATA_PRODUCT_DETAILS,
+  SET_DATA_PRODUCT_DETAILS
+} from "../constants";
+import apiFetch from "../utils/apiFetch";
+
+export const getProduct = idProduct => {
+  return async dispatch => {
+    dispatch({ type: START_DATA_PRODUCT_DETAILS });
+    const data = await apiFetch(`/products/${idProduct}`);
+    if (data.status === 200) {
+      dispatch({ type: END_DATA_PRODUCT_DETAILS });
+    }
```

Comment:
also you can set case if data.status is not 200, so request returned with error you can catch them like this

```
if (data.status === 200) {
      dispatch({ type: END_DATA_PRODUCT_DETAILS });
    } else {
return  dispatch({ type: ERROR_DATA_PRODUCT_DETAILS });

}
```